core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -13,86 +13,186 @@
+                                             .collect(Collectors.toList());
+
+            if (REGEXP_LIKE_FUNCTION_NAME.equalsIgnoreCase(call.getName()) && arguments.size() == 2) {
+                Expression pattern = arguments.get(1);
+                if (pattern instanceof StringLiteral) {
+                    Slice slice = ((StringLiteral) pattern).getSlice();
+                    if (RegexLibrary.JONI.equals(SystemSessionProperties.getRegexLibrary(session))) {
+                        JoniRegexp joniRegexp = JoniRegexpCasts.castVarcharToJoniRegexp(slice);
+                        arguments.set(1, literalEncoder.toExpression(joniRegexp, JoniRegexpType.JONI_REGEXP));
+                    }
+                    else if (RegexLibrary.RE2J.equals(SystemSessionProperties.getRegexLibrary(session))) {
+                        int dfaStatesLimit = SystemSessionProperties.getRe2JDfaStatesLimit(session);
+                        int dfaRetries = SystemSessionProperties.getRe2JDfaRetries(session);
+                        Re2JRegexp re2JRegExp = new Re2JRegexp(dfaStatesLimit, dfaRetries, slice);
+                        Type type = new Re2JRegexpType(dfaStatesLimit, dfaRetries);
+                        arguments.set(1, literalEncoder.toExpression(re2JRegExp, type));
+                    }
+                }
             }
 
-            if (expression instanceof FieldDereference) {
-                FieldDereference dereference = (FieldDereference) expression;
-                return new SubscriptExpression(translate(dereference.getTarget()), new LongLiteral(Long.toString(dereference.getField() + 1)));
+            return Optional.of(new FunctionCall(Optional.empty(),
+                                                qualifiedName,
+                                                Optional.empty(),
+                                                Optional.empty(),
+                                                Optional.empty(),
+                                                false,
+                                                Optional.empty(),
+                                                Optional.empty(),
+                                                arguments));
+        }
+
+        protected Optional<Expression> translateLike(ConnectorExpression value, ConnectorExpression pattern)
+        {
+            Optional<Expression> valueExpression = process(value);
+            Optional<Expression> patternExpression = process(pattern);
+            if (valueExpression.isPresent() && patternExpression.isPresent()) {
+                return Optional.of(new LikePredicate(valueExpression.get(), patternExpression.get(), Optional.empty()));
             }
+            return Optional.empty();
+        }
 
-            throw new UnsupportedOperationException("Expression type not supported: " + expression.getClass().getName());
+        private QualifiedName getQualifiedName(Call call)
```

Comment:
This is unnecessary. The `Call` should already indicate a fully qualified name. See comment above.

Reply:
I'm not sure how, see the above response please

Reply:
Done